tests/suites/os/tests/safe-reboot/index.js

```diff
@@ -0,0 +1,194 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const request = require('request-promise');
+const SUPERVISOR_PORT = 48484;
+
+module.exports = {
+	title: 'Safe reboot tests',
+	tests: [
+		{
+			title: 'Respect application locks',
+			run: async function(test) {
+				let fd = '';
+				let result = '';
+				fd = await this.worker.executeCommandInHostOS(
+					`touch /tmp/reboot-check && \
+					mkdir -p /tmp/balena-supervisor/services/test/ && \
+					touch /tmp/balena-supervisor/services/test/updates.lock && \
+					exec {FD}</tmp/balena-supervisor/services/test/updates.lock && \
+					(flock -n $FD && sleep infinity) &
+					echo "$FD"`,
+					this.link,
+				)
+				await this.worker.executeCommandInHostOS(
+					'/usr/libexec/safe_reboot &',
```

Comment:
Does executing background tasks and disconnecting the session actually retain the task? I thought things like `screen` were needed for that to work.

If not, these might be better performed by async node functions with (or without) callbacks to keep the sessions alive.